src/bridge/cockpitpcpmetrics.c

```diff
@@ -47,14 +47,24 @@ typedef struct {
 } MetricInfo;
 
 typedef struct {
+  int context;
+  gint64 start;
+} ArchiveInfo;
```

Comment:
Given that you'll probably have at least two archives (one recent and one currently-written-to), this sort of gluing-together would be helpful & nice.  PMAPI will provide exactly this service within a month or two.  (It doesn't mean that individual archive boundaries must necessarily be communicated to the gui.)

pmmgr can keep all prior (but age-limited) data within a single archive (in addition to the currently-written-to one), but it turns out that repeatedly merging/slicing archives together into one tickles a libpcp bug http://oss.sgi.com/bugzilla/show_bug.cgi?id=1044 so we've switched defaults to 'granular' mode, so that N period-aligned archives are kept)